src/ipc/ipc-helper.c

```diff
@@ -36,6 +36,15 @@
 
 LOG_MODULE_DECLARE(ipc, CONFIG_SOF_LOG_LEVEL);
 
+static bool valid_ipc_buffer_desc(const struct sof_ipc_buffer *desc)
+{
+	if (desc->caps >= SOF_MEM_CAPS_MAX_BIT)
+		return false;
```

Comment:
`return desc->caps < SOF_MEM_CAPS_MAX_BIT;`

Reply:
Thanks but there is a TODO on the next line.

Anyway I don't like this PR for other reasons mentioned above.

Reply:
Rework PR or close ?

Reply:
I will give this another try. Keeping as a draft for now.